source/interprocedural_analyses/taint/stepLogger.ml

```diff
@@ -0,0 +1,24 @@
+open Core
+
+module StepLogger = struct
+  type t = {
+    start_message: string;
+    end_message: string;
+    timer: Timer.t;
+  }
+
+  let start ~start_message ~end_message =
+    let timer = Timer.start () in
+    Log.info "%s..." start_message;
+    { start_message; end_message; timer }
+
+  let finish ~step_logger ~integers () =

```

Comment:
not a big deal, but just to adopt a convention:
Usually, a module that defines a `type t` and functions on that type will take that parameter as the last (or sometimes first) parameter (and without using named parameters).
For instance, you wouldn't use `List.concat ~list` or `List.append ~list ~element`. That would break a lot of potential for chaining.

TL, DR: I would suggest something like:
```suggestion
  let finish ~integers step_logger =
```
This also gets rid of the `()`, which is a nice benefit.